### PR TITLE
Feature/add archiver

### PR DESCRIPTION
**Pull request by bspindler**

### CODE REVIEW

File: ts/commands/PackageCommands.ts [hunk 1]
```diff
@@ -21,6 +21,14 @@ class PackageCommands {
         packageValidator.validate(location, require(location + '/package.json').id);
       });
 
+    (caporal as any)
+    .command('package create', 'Create a local pkg.zip')
+    .action((args, options, logger) => {
+      const location: string = path.resolve(process.cwd());
+      const config = configService.mergeConfig(options);
+      packageService.createArchive(location, config, logger);
```
Comment (TheConnMan) [inline ts/commands/PackageCommands.ts]: Location is undefined here.

File: ts/services/PackageService.ts [hunk 2]
```diff
@@ -102,6 +103,50 @@ class PackageService {
...
+    logger.debug('\nCreating archive package ');
+    const output = fs.createWriteStream(location + '/pkg.zip');
+    const archive = archiver('zip', {
+      zlib: { level: 9 } // Sets the compression level.
+    });
+
+    // listen for all archive data to be written
+    // 'close' event is fired only when a file descriptor is involved
+    output.on('close', () =>  {
+      logger.info(archive.pointer() + ' total bytes');
```
Comment (TheConnMan) [inline ts/services/PackageService.ts]: This feels like more of a debug log.

File: ts/services/PackageService.ts [hunk 3]
```diff
@@ -102,6 +103,50 @@ class PackageService {
...
+
+    // good practice to catch this error explicitly
+    archive.on('error', (err) =>  {
+      throw err;
+    });
+
+    // pipe archive data to the file
+    archive.pipe(output);
+
+    archive.directory('./', 'pkg-dir');
```
Comment (TheConnMan) [inline ts/services/PackageService.ts]: Should this zip up the `location` directory?

File: ts/services/PackageService.ts [hunk 4]
```diff
@@ -102,6 +103,50 @@ class PackageService {
...
+    archive.finalize();
+
```
Comment (TheConnMan) [inline ts/services/PackageService.ts]: Trying to find a TSLint rule to catch this trailing new line in a function, but can't find one.

File: ts/commands/PackageCommands.ts (hunk 1 above)
Comment (bspindler) [reply ts/commands/PackageCommands.ts]: I can log it and it's defined for me.  It's the current working directory.

File: ts/services/PackageService.ts (hunk 2 above)
Comment (bspindler) [reply ts/services/PackageService.ts]: done

File: ts/commands/PackageCommands.ts (hunk 1 above)
Comment (bspindler) [reply ts/commands/PackageCommands.ts]: yeah, there is no --location option, I suppose I could add one.

File: ts/services/PackageService.ts (hunk 3 above)
Comment (bspindler) [reply ts/services/PackageService.ts]: yes, good catch

File: ts/services/PackageService.ts (hunk 4 above)
Comment (bspindler) [reply ts/services/PackageService.ts]: I removed it

**Pull request merged**